rust/lk/src/log.rs

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2024 Google Inc. All rights reserved
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction,
+ * including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+// TODO: replace with `trusty-log` crate once it is `no_std`-compatible
+
+use alloc::ffi::CString;
+use alloc::format;
+use core::ffi::c_uint;
+use log::{LevelFilter, Log, Metadata, Record};
+
+use crate::LK_INIT_HOOK;
+use crate::init::lk_init_level;
+
+use crate::lk_stderr;
+use crate::sys::fflush;
+use crate::sys::fputs;
+// use crate::sys::LK_LOGLEVEL_RUST;
+// TODO: Configure this properly.
+static LK_LOGLEVEL_RUST: usize = 4;
+
+static TRUSTY_LOGGER: TrustyKernelLogger = TrustyKernelLogger;
```

Comment:
I wonder if we should avoid mentioning Trusty here, as LK is really its own project.